models/attachment.go

```diff
@@ -138,19 +138,22 @@ func DeleteAttachment(a *Attachment, remove bool) error {
 
 // DeleteAttachments deletes the given attachments and optionally the associated files.
 func DeleteAttachments(attachments []*Attachment, remove bool) (int, error) {
+	if len(attachments) == 0 {
+		return 0, nil
+	}
+
+	var ids = make([]int64, 0, len(attachments))
 	for i, a := range attachments {
 		if remove {
 			if err := os.Remove(a.LocalPath()); err != nil {
 				return i, err
```

Comment:
In this case, there are some attachments which we have already deleted from the filesystem. Should we also remove them from the DB, so that the DB doesn't contain non-existent attachments?

Reply:
additional check could be added to check if error is about nonexistent file but if error is that file can not be deleted (access denied etc) than it should stay as it is now